New web platform release notes for 152 (#3889)
MicrosoftDocs/edge-developer: microsoft-edge/web-platform/release-notes/152.md

  </template>
</custom-checkbox>
```

See also:
* [`referencetarget`](https://developer.mozilla.org/docs/Web/API/Element/attachShadow#referencetarget), in _Element: `attachShadow()` method_, at MDN.
* [`Shadowroot.referencetarget`](https://developer.mozilla.org/docs/Web/API/ShadowRoot#shadowroot.referencetarget), in _`ShadowRoot`_, at MDN.
* [`shadowrootreferencetarget`](https://developer.mozilla.org/docs/Web/HTML/Reference/Elements/template#shadowrootreferencetarget), in _`<template>` HTML content template element_, at MDN.

<!-- ------------------------------ -->
#### `autocorrect` global html attribute

The `autocorrect` global HTML attribute controls whether to automatically correct spelling or punctuation errors for user input in `<input>` and `<textarea>` elements, and elements with the `contenteditable` attribute.

See also:
* [autocorrect HTML global attribute](https://developer.mozilla.org/docs/Web/HTML/Reference/Global_attributes/autocorrect) at MDN.

<!-- ====================================================================== -->
## Web APIs

The following new Web API features are included in Microsoft Edge.

<!-- ------------------------------ -->

> #### Planned deprecation of the unload event

Reviewer: is the deprecation section correctly placed in the outline?

https://review.learn.microsoft.com/en-us/microsoft-edge/web-platform/release-notes/152?branch=pr-en-us-3889#planned-deprecation-of-the-unload-event

Author: Yes, that seems fine to me. It's a headsup at this point, we don't need a dedicated "Deprecated features" section or similar. Starting with 153, we might want to create such a section.

Reviewer: closing.  there's a todo, might keep for future:
`<!-- todo: repeat this section in webplatform relnotes 153, 154, 155 -->`

<!-- todo: repeat this section in webplatform relnotes 153, 154, 155 -->

> Edge 153 through 155 introduces a new Permission-Policy to allow creating unload event listeners.  The default policy is `allow`, but the default policy will gradually be migrated to `deny`, such that unload handlers stop firing on pages, unless a page explicitly opts in to re-enable unload handlers.

Author: ```suggestion
Edge 153 will introduce the new `unload` Permission-Policy and gradually set its default value to `deny` between Edge 153 and Edge 155. To explicitely enable unload event handlers, use the `allow` value.
```

Reviewer: explicitly

> * Edge 153: Unload handlers will stop firing by default for 60% of page loads, unless the site explicitly enables unload handlers via Permissions Policy.

Reviewer: ```suggestion
* Edge 153: Unload handlers will stop firing by default for 60% of page loads, unless the site explicitly enables unload handlers via the `unload` Permissions Policy.
```
fyi copied from table cell @captainbrosset

* [`Shadowroot.referencetarget`](https://developer.mozilla.org/docs/Web/API/ShadowRoot#shadowroot.referencetarget), in _`ShadowRoot`_, at MDN.
* [`shadowrootreferencetarget`](https://developer.mozilla.org/docs/Web/HTML/Reference/Elements/template#shadowrootreferencetarget), in _`<template>` HTML content template element_, at MDN.

<!-- ------------------------------ -->
#### `autocorrect` global html attribute

The `autocorrect` global HTML attribute controls whether to automatically correct spelling or punctuation errors for user input in `<input>` and `<textarea>` elements, and in elements with the `contenteditable` attribute.

See also:
* [autocorrect HTML global attribute](https://developer.mozilla.org/docs/Web/HTML/Reference/Global_attributes/autocorrect) at MDN.

<!-- ====================================================================== -->
## Web APIs

The following new Web API features are included in Microsoft Edge.

<!-- ------------------------------ -->
#### Planned deprecation of the unload event
<!-- todo: repeat this section in webplatform relnotes 153, 154, 155 -->

> Edge 153 will introduce the new `unload` Permission-Policy, to allow creating unload event listeners.  The default value is initially `allow`, but between Edge 153 and Edge 155, the default value will gradually be migrated to `deny`, such that unload handlers stop firing on pages, unless a page explicitly opts in to re-enable them.  To explicitly enable unload event handlers, use the `allow` value.

Reviewer 2: ```suggestion
Edge 152 will introduce the new `unload` Permission-Policy, to allow creating unload event listeners.  The default value is initially `allow`, but between Edge 152 and Edge 155, the default value will gradually be migrated to `deny`, such that unload handlers stop firing on pages, unless a page explicitly opts in to re-enable them.  To explicitly enable unload event handlers, use the `allow` value.
```

Reviewer: & site-compat table cell

> * Edge 153: Unload handlers will stop firing by default for 60% of page loads, unless the site explicitly enables unload handlers via the `unload` Permissions Policy.

Reviewer 2: ```suggestion
* Edge 152 and 153: Unload handlers will stop firing by default for 60% of page loads, unless the site explicitly enables unload handlers via the `unload` Permissions Policy.
```

* Edge 154: Increases to 80% of page loads.
* Edge 155: Increases to 100% of page loads.

This change is happening in the Chromium project, on which Microsoft Edge is based.

The unload event isn't a dependable way to do the following, because browsers and operating systems can suspend or terminate pages without firing the unload event:
* Save data.
* Perform cleanup.
* Report the end of a session.

> * Using the unload event can also prevent pages from using the back/forward cache, which might make back and forward navigation slower.

Author: ```suggestion

Using the unload event can also prevent pages from using the back/forward cache, which might make back and forward navigation slower.
```

Reviewer: done

preserved improvement: 
the sentence remains sep. para from "the above scenarios" sentence, which prevents concat'ing the sentences.

* [Element: pseudo() method](https://developer.mozilla.org/docs/Web/API/Element/pseudo) at MDN.
* [CSSPseudoElement: pseudo() method](https://developer.mozilla.org/docs/Web/API/CSSPseudoElement/pseudo) at MDN.

<!-- ------------------------------ -->
#### Connection allowlists

Connection allowlists let you declare and enforce explicit allowlists for network endpoints that can be reached by networking APIs such as the Fetch API.  This improves outbound network control, and reduces the risk of unintended connections.

The list of authorized endpoints is distributed from the server through the `Connection-Allowlist` HTTP response header.  Before connecting to a server, the browser evaluates the destination against this allowlist.  Connections to verified endpoints are permitted, but destinations that fail to match entries in the allowlist are blocked.

See also:
* [Connection Allowlists](https://github.com/WICG/connection-allowlists)
* [Fetch API](https://developer.mozilla.org/docs/Web/API/Fetch_API) at MDN.

<!-- ------------------------------ -->
#### Notification attribution for PWAs on macOS

> <!-- todo: wait for confirmation from the PWA team that this applies to Edge too -->

Reviewer: Notification attribution for PWAs on macOS - wait for confirmation from PWA team that this applies to Edge

@captainbrosset

Author: Can we hide the section for the time being (hidden within an HTML comment block)? I haven't heard back yet, but don't want to delay publishing the release notes for too long.
When/if I hear, we can add the section. But for now, let's default to hiding it.

Reviewer: commented out h4 section

> When a Progressive Web App (PWA) is installed on macOS, the notifications that are sent by the PWA are now attributed to the PWA instead of Microsoft Edge.  The notification now shows the PWA name and icon in the Notification Center, instead of showing the Microsoft Edge name and icon.

Author: ```suggestion
When a Progressive Web App (PWA) is installed on macOS, the notifications that are sent by the PWA are now attributed to the PWA instead of to Microsoft Edge.  The notification now shows the PWA name and icon in the Notification Center, instead of showing the Microsoft Edge name and icon.
```

To better match native macOS notifications, the following changes were also made:

* Microsoft Edge no longer supports the `requireInteraction` field for notifications.  On macOS, the user controls whether the notification is temporary or persistent, on a per-app basis.
* The Badging API now requires notifications permissions for the app badge to be displayed.  If the user doesn't grant notifications permission, the API will do nothing.

See also:
* [Notifications API](https://developer.mozilla.org/docs/Web/API/Notifications_API) at MDN.
* [Notification: requireInteraction property](https://developer.mozilla.org/docs/Web/API/Notification/requireInteraction) at MDN.
* [Badging API](https://developer.mozilla.org/docs/Web/API/Badging_API) at MDN.

<!-- ------------------------------ -->
#### `OpaqueRange` for form control text

Use `OpaqueRange` to represent a live span of text within a form control's value, such as a `<textarea>` or text-based `<input>`.

`OpaqueRange` enables the following for inline suggestions, highlights, and anchored popovers:
* Operations such as `getBoundingClientRect()` and `getClientRects()`.

> * Integration with the CSS Custom Highlight API.

Author: ```suggestion
`OpaqueRange` enables to do the following with  ranges of text:
* Operations such as `getBoundingClientRect()` and `getClientRects()`.
* Integration with the CSS Custom Highlight API.

This enables scenarios such as inline suggestions, highlights, and anchored popovers.
```

The `getDisplayMedia()` method now supports the `audioSelection:preferred` option.

This option allows you to indicate to the browser that you prefer audio sharing along with video.

See also:
* [MediaDevices: getDisplayMedia() method](https://developer.mozilla.org/docs/Web/API/MediaDevices/getDisplayMedia) at MDN.

<!-- ====================================================================== -->
## Origin trials

The following are origin trials for new experimental APIs that are available in Microsoft Edge.

Origin trials let you try experimental APIs on your own live website for a limited time.  To learn more about origin trials, see [Use origin trials in Microsoft Edge](../../origin-trials/index.md).

For the full list of available origin trials, see [Microsoft Edge Origin Trials](https://developer.microsoft.com/microsoft-edge/origin-trials/).

| Name | Description | Register |
|---|---|---|
| Soft navigation heuristics | Exposes soft-navigation heuristics for collecting performance metrics in single-page apps. | [Register](https://developer.microsoft.com/microsoft-edge/origin-trials/trials/ec952482-49f0-4980-a451-13ea3f9a7220) |

> | WebNN [Closed] | <!-- todo: 150 cell has only, instead: Build and execute hardware-accelerated machine-learning models directly in your web app. --> [This trial is no longer accepting new registrations.]  WebNN provides a high-level abstraction over the ML acceleration capabilities provided by the platform.  The goal of this experiment is to understand how well real-world models abstract to the operations supported by WebNN, and how well WebNN can map these operations to operations that are supported by the hardware of real-world users. | [Register](https://developer.microsoft.com/microsoft-edge/origin-trials/trials/19857284-cf52-484b-8b09-8ca50ac9dccb) |

Reviewer: WebNN - 150's cell has shorter text

Author: Let's delete this row, the OT is closed.

| WebAssembly Custom Descriptors V2 | Stores data that's associated with source-level types more efficiently, in custom descriptor objects. | [Register](https://developer.microsoft.com/microsoft-edge/origin-trials/trials/9c75e7f3-49cb-43de-9745-7458ff9c3689) |
| Speculative Load measurement | `performance.speculations` exposes data about used and unused preloads, prefetches, and prerenders.  This helps you measure the effectiveness of speculative loading strategies and optimize performance-versus-cost tradeoffs. | [Register](https://developer.microsoft.com/microsoft-edge/origin-trials/trials/246de200-45d2-4f44-8047-001d385c90c5) |
| Additional WebCrypto algorithms | Enables usage of additional WebCrypto algorithms, including a few PQ algorithms.  Algorithms enabled: ChaCha20-Poly1305, ML-DSA-44, ML-DSA-65, ML-DSA-87, ML-KEM-768, ML-KEM-1024, MLKEM768-X25519. | [Register](https://developer.microsoft.com/microsoft-edge/origin-trials/trials/f5dbdb03-9776-4ee2-9560-4d68733c7531) |

> | WebRTC Data Channel: SCTP Negotiation Acceleration Protocol (SNAP) | <!-- new, todo: condense --> WebRTC Data Channels use the Stream Control Transmission Protocol (SCTP) over a Datagram Transport Layer Security (DTLS) association.  The standard SCTP connection establishment requires a handshake that introduces latency.  A new Internet draft specifies a method to accelerate establishing the data channel by embedding the SCTP initialization parameters within the Session Description Protocol (SDP) offer/answer exchange.  This reduces the time required to open a data channel by up to two network round-trip times. | [Register](https://developer.microsoft.com/microsoft-edge/origin-trials/trials/fd803c89-97ff-4d0d-bc23-641b6041cb12) |

Reviewer: WebRTC Data Channel - condense

Author: ```suggestion
| WebRTC Data Channel: SCTP Negotiation Acceleration Protocol (SNAP) | SNAP reduces WebRTC data channel startup latency by embedding SCTP connection setup parameters in the SDP offer/answer exchange. Because WebRTC already uses DTLS to establish a secure channel, the separate SCTP handshake can be avoided, saving up to two network round trips and enabling faster data channel establishment. | [Register](https://developer.microsoft.com/microsoft-edge/origin-trials/trials/fd803c89-97ff-4d0d-bc23-641b6041cb12) |
```
